chrome/browser/ui/webui_browser/webui_browser_browsertest.cc

```diff
@@ -84,3 +86,324 @@ IN_PROC_BROWSER_TEST_F(WebUIBrowserTest, MAYBE_NavigatePage) {
   EXPECT_EQ("Default response given for path: /defaultresponse",
             EvalJs(web_contents, "document.body.textContent"));
 }
+
+#if BUILDFLAG(IS_CHROMEOS)
+// TODO(crbug.com/451876195): Fix and re-enable this test for CrOS.
+// For now this is disabled on CrOS since BrowserStatusMonitor/
+// AppServiceInstanceRegistryHelper aren't happy with our shutdown deletion
+// order of native windows vs. Browser and aren't tracking the switch over
+// of views on child guest contents properly.
+#define MAYBE_EnumerateDevToolsTargets DISABLED_EnumerateDevToolsTargets
+#else
+#define MAYBE_EnumerateDevToolsTargets EnumerateDevToolsTargets
+#endif
+// Verify DevTools targets enumeration for browser UI and tabs.
+IN_PROC_BROWSER_TEST_F(WebUIBrowserTest, MAYBE_EnumerateDevToolsTargets) {
+  auto* window = browser()->window();
+  ASSERT_TRUE(window);
+
+  content::WebContents* web_contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  ASSERT_TRUE(web_contents);
+  EXPECT_TRUE(content::WaitForLoadStop(web_contents));
+
+  // Make sure that the web contents actually got converted to a guest and in
+  // DOM before enumerate DevTools targets.
+  EXPECT_TRUE(base::test::RunUntil([web_contents]() {
+    return web_contents->GetOuterWebContents() != nullptr;
+  }));
+
+  // Verify DevTools target types.
+  auto targets = content::DevToolsAgentHost::GetOrCreateAll();
+  int tab_count = 0;
+  int page_count = 0;
+  int browser_ui_count = 0;
+  auto hosts = content::DevToolsAgentHost::GetOrCreateAll();
+  for (auto& host : hosts) {
+    LOG(INFO) << "Found DevTools target, type: " << host->GetType()
+              << ", parent id:" << host->GetParentId()
+              << ", url: " << host->GetURL().spec();
+    // Only expect top level targets.
+    EXPECT_TRUE(host->GetParentId().empty());
+    if (host->GetType() == content::DevToolsAgentHost::kTypeTab) {
+      ++tab_count;
+    } else if (host->GetType() == content::DevToolsAgentHost::kTypePage) {
+      ++page_count;
+    } else if (host->GetType() == content::DevToolsAgentHost::kTypeBrowserUI) {
+      ++browser_ui_count;
+    }
+  }
+  // Expect browser_ui target for browser UI main frame, Tab target for tab
+  // WebContents, and Page target for tab main frame.
+  EXPECT_EQ(hosts.size(), 3U);
+  EXPECT_EQ(browser_ui_count, 1);
+  EXPECT_EQ(tab_count, 1);
+  EXPECT_EQ(page_count, 1);
+}
+
+#if !BUILDFLAG(IS_CHROMEOS)
+// Begin security related tests. These tests validate the security
+// boundary between a GuestContents and the parent.
+class WebUIBrowserSecurityTest : public WebUIBrowserTest {
```

Comment:
This WebUIBrowserSecurityTest was not really part of the original change, and was later removed. We probably want to remove it from this CP.

Reply:
Removed, thanks!